pkg/util/port_split.go

```diff
@@ -38,3 +41,19 @@ func SplitPort(id string) (name, port string, valid bool) {
 	}
 	return id, "", len(id) > 0
 }
+
+// ParsePort parses a raw host string of the form "host:port".
+// Will error if port is absent, non-numeric, or out of allowed port range.
+func ParsePort(rawhost string) (host string, port int, err error) {
+	var portString string
+	if host, portString, err = net.SplitHostPort(rawhost); err != nil {
+		return "", 0, fmt.Errorf("failed to parse port from host (%s): %v", rawhost, err)
+	}
+	if port, err = strconv.Atoi(portString); err != nil {
+		return "", 0, fmt.Errorf("failed to parse port as int from host (%s): %v", rawhost, err)
+	}
+	if port < 0 || port > 0xFFFF {
```

Comment:
Kinda depends on the usage, actually. Usually zero means "give me a free port, let the os decide".